src/main/java/kitchenpos/menu/tobe/domain/menu/MenuProducts.java

```diff
@@ -0,0 +1,57 @@
+package kitchenpos.menu.tobe.domain.menu;
+
+import jakarta.persistence.*;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+
+@Embeddable
+public class MenuProducts {
+    @OneToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
+    @JoinColumn(
+            name = "menu_id",
+            nullable = false,
+            columnDefinition = "binary(16)",
+            foreignKey = @ForeignKey(name = "fk_menu_product_to_menu")
+    )
+    private List<MenuProduct> menuProducts;
+
+    protected MenuProducts() {
+    }
+
+    public MenuProducts(List<MenuProduct> menuProducts, BigDecimal menuPrice) {
+        if (Objects.isNull(menuProducts) || menuProducts.isEmpty()) {
+            throw new IllegalArgumentException("메뉴에 속한 상품이 비어있습니다.");
+        }
+        this.menuProducts = menuProducts;
+        validateMenuPriceDoesNotExceedTotalProductPrice(menuPrice);
```

Comment:
사소하지만 유효성 검증을 먼저 진행하도록해주면 불필요한 인스턴스 생성을 막을 수 있을 것 같네요. 🙂

```suggestion
        validateMenuPriceDoesNotExceedTotalProductPrice(menuPrice);
        this.menuProducts = menuProducts;
```

Reply:
이 부분은 `validateMenuPriceDoesNotExceedTotalProductPrice()`에서 방금 주입받은 `menuProducts`를 검증하는 부분이 들어가서 순서상 어쩔 수 없이 위와 같이 작성했습니다!

Reply:
음... 말씀하신 문제는 

단순히 `validateMenuPricePolicy` 메서드 파라미터로 생성자 (인스턴스 필드가 아니라) 파라미터로 전달받은 `menuProducts`를 주입해주면 될 것 같은데 혹시 다른 문제점이 있을까요? 🙂
```java
        validateMenuPricePolicy(menuProducts, menuPrice);
        this.menuProducts = menuProducts;
```

Reply:
음... 말씀하신 문제는 

단순히 `validateMenuPricePolicy` 메서드 파라미터로 생성자 (인스턴스 필드가 아니라) 파라미터로 전달받은 `menuProducts`를 주입해주면 될 것 같은데 혹시 다른 문제점이 있을까요? 🙂
```java
        validateMenuPricePolicy(menuProducts, menuPrice);
        this.menuProducts = menuProducts;
```